daffodil-lib/src/main/scala/org/apache/daffodil/lib/util/DecimalUtils.scala

```diff
@@ -386,6 +386,25 @@ object DecimalUtils {
     outArray
   }
 
+  /**
+   * Despite the name, this does not actually convert the digit from EBCDIC encoding.
+   * The characters have already been decoded from whatever charset into Unicode code points.
+   * What this does is deal with overpunched negative sign character mappings.
+   * In this case "}JKLMNOPQR" are negative 0 to 9.
+   * @param digit
+   * @return A pair of the integer the digit represents, and a boolean indicating if negative.
+   */
+  def convertFromZonedEBCDIC(digit: Char): (Int, Boolean) = {
+    if ((digit >= '0') && (digit <= '9')) // positive 0-9
+      (digit - 48, false)
+    else if (digit == '}') // negative 0
+      (0, true)
+    else if ((digit >= 'J') && (digit <= 'R')) // negative 1-9
+      (digit - 'J' + 1, true)
+    else
+      throw new NumberFormatException("Invalid zoned digit: " + digit)
```

Comment:
The DFDL spec says this about `textNumberRep`:
> In EBCDIC-based encodings, code points 0xC0 to 0xC9 or 0xF0 to 0xF9 represent a positive sign and digits 0 to 9 (these byte ranges correspond typically to characters '{ABCDEFGHI'* or '0123456789'), and code points 0xD0 to 0xD9 or 0xB0 to 0xB9 represent a negative sign and digits 0 to 9 (these byte ranges correspond typically to characters '}JKLMNOPQR' or  '^£¥·©§¶¼½¾ ' ). On parsing both ranges are accepted. On unparsing the range 0xC0 to 0xC9 are produced for positive signs and the range 0xD0 to 0xD9 are produced for negative signs.

Should this function also support `{ABCDEFGHI` for positive numbers and `^£¥·©§¶¼½¾ ` for negatives?

Reply:
Correct. Fixed.